Tests/js/views/services/plugins/assets/ez-locationswapplugin-tests.js

```diff
@@ -0,0 +1,623 @@
+                    }
+                }
+            });
+
+            return locationMock;
+        },
+
+        _getContentInfoMock: function (attrs) {
+            var contentInfoMock = new Mock();
+
+            Mock.expect(contentInfoMock, {
+                'method': 'get',
+                'args': [Mock.Value.String],
+                'run': function (attr) {
+                    switch (attr) {
+                        case 'id':
+                            return attrs.id;
+                        case 'name':
+                            return attrs.name;
+                        case 'mainLanguageCode':
+                            return attrs.languageCode;
+                        default:
+                            Assert.fail('Trying to `get` incorrect attribute');
+                            break;
+                    }
+                }
+            });
+
+            return contentInfoMock;
+        },
+
+        _getSelection: function () {
+            this.destinationContent = {'id': '/content/Fernando/Torres', 'name': 'Fernando Torres'};
+            this.destinationContentInfo = this._getContentInfoMock(this.destinationContent);
+            this.destinationLocation = {'id': '/location/Fernando/Torres', 'contentInfo': this.destinationContentInfo};
+
+            return {location: this._getLocationMock(this.destinationLocation)};
+        },
+
+        "Should swap locations and fire notifications": function () {
+            var startNotificationFired = false,
+                successNotificationFired = false,
+                errorNotificationFired = false,
+                appNavigateCalled = false,
+                that = this;
+
+            this.service.set('app', {
+                navigateTo: function (routeName, params) {
+                    appNavigateCalled = true;
+
+                    Assert.areEqual('viewLocation', routeName, 'The route name should be `viewLocation`');
+                    Assert.areEqual(
+                        params.id,
+                        that.locationJson.id,
+                        'Id of location of content should be passed as id param'
+                    );
+                    Assert.areEqual(
+                        params.languageCode,
+                        that.contentJson.languageCode,
+                        'Main language code of content should be passed as language param'
+                    );
+                }
+            });
+
+            this.service.on('contentDiscover', function (e) {
+                var config = {
+                    selection: that._getSelection(),
+                    target: {
+                        get: function () {}
+                    }
+                };
+
+                e.config.contentDiscoveredHandler(config);
+            });
+
+            Mock.expect(this.location, {
+                'method': 'swap',
+                'args': [Mock.Value.Object, Mock.Value.Object, Mock.Value.Function],
```

Comment:
for the second argument, you should either give a real value (the Location that was selected) or add an `Assert.areSame` call in the run function to make sure the correct Location is used there.